### PR TITLE
User component

### LINKED ISSUE #51: FE: Users components

### PR DESCRIPTION
**Pull request by Junjpan**

### CODE REVIEW

File: src/routes/users.js [hunk 1]
```diff
@@ -121,12 +131,7 @@ router
 
         user.save().then(user => {
           const { id, email, name, userImage } = user;
-          res.status(201).json({
-            id,
-            email,
-            name,
-            userImage,
-          });
+          res.status(201).send('success');
```
Comment (rusye) [inline src/routes/users.js:134]: So would you be refreshing the component on the frontend and doing another GET request?  I was thinking you could potentially grab everything from that response and use it in that state?  Just a thought.

File: client/src/components/main/sidebar/Sidebar.js [hunk 2]
```diff
@@ -55,10 +56,42 @@ const Active = styled.i`
...
+    axios
+      .get(`/api/users/${localStorage.userId}`, {
+        headers: { authorization: `bearer ${localStorage.authToken}` },
+      })
+      .then(res => {
+        setLogedinUser(res.data.name);
+        setImageUrl(res.data.userImage);
+      })
+      .catch(err => console.error('Unable to get the user'));
+  }, [logedinUser, imageUrl]);
```
Comment (rusye) [inline client/src/components/main/sidebar/Sidebar.js:94]: We could potentially remove all this, and when a user logs in we can set the name and image to the local storage that way we have one less call to the server.

File: client/src/components/main/sidebar/user/Image.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+`;
+
+function Image(props) {
+  const [url, setUrl] = useState('');
+
+  useEffect(() => {
+    createJdention(props.image, props.id, 'bigimage');
```
Comment (rusye) [inline client/src/components/main/sidebar/user/Image.js:28]: We could probably incorporate this into the comment component as well.

File: client/src/components/main/sidebar/user/Image.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+          </SmallButton>
+
+          <SmallButton
+            type="button"
+            background="white"
+            onClick={deleteImage}
+            disabled={props.imageurl === null ? true : false}
+          >
+            Delete
+          </SmallButton>
```
Comment (rusye) [inline client/src/components/main/sidebar/user/Image.js:125]: Do you think we should have an alert pop up asking if they really want to do that move?

File: client/src/components/main/sidebar/user/Image.js [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+            Preview
+          </SmallButton>
+
+          <SmallButton type="button" onClick={resetImage} disabled={url.length >= 1 ? false : true}>
+            Reset
+          </SmallButton>
+
+          <SmallButton type="submit" background="white">
+            Update
+          </SmallButton>
```
Comment (rusye) [inline client/src/components/main/sidebar/user/Image.js:116]: Maybe add more info on the frontend to the user to what these buttons do?  I had to assume what these buttons do.

File: client/src/components/main/sidebar/user/User.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  return (
+    <Div ref={container}>
+      <Profile>
+        <ProfileImage id="smallimage"></ProfileImage>
+
+        <Name>
+          <b>
+            {!changeUserName ? (
+              userstate.username
+            ) : (
```
Comment (rusye) [inline client/src/components/main/sidebar/user/User.js:105]: Unable to click outside the input to make it cancel the username change

File: client/src/components/main/sidebar/user/Image.js (hunk 5 above)
Comment (Junjpan) [reply client/src/components/main/sidebar/user/Image.js:116]: In a real world, there are not too many instructions to tell which button to do what. A lot of times users need to figure themselves and I think the buttons here are pretty straight forward.

File: client/src/components/main/sidebar/user/Image.js (hunk 4 above)
Comment (Junjpan) [reply client/src/components/main/sidebar/user/Image.js:125]: I can do that

File: client/src/components/main/sidebar/Sidebar.js (hunk 2 above)
Comment (Junjpan) [reply client/src/components/main/sidebar/Sidebar.js:94]: Yes, we could potentially do that. But is username  sensitive info for user?

File: src/routes/users.js (hunk 1 above)
Comment (Junjpan) [reply src/routes/users.js:134]: As long as I receive the  success message from the back end, I can just use all the information from the front end.

File: client/src/components/main/sidebar/user/User.js (hunk 6 above)
Comment (Junjpan) [reply client/src/components/main/sidebar/user/User.js:105]: I can fix it.

**Review by rusye — APPROVED** [review]

**Review by Igor333m — APPROVED** [review]

**Pull request merged**